pkg/controllers/nodeclaim/disruption/controller.go

```diff
@@ -125,6 +129,15 @@ func (c *Controller) Builder(_ context.Context, m manager.Manager) operatorcontr
 		Watches(
 			&v1.Pod{},
 			nodeclaimutil.PodEventHandler(c.kubeClient),
-		),
-	)
+		)
+	for _, ncGVK := range c.supportedNodeClasses {
+		nodeclass := &unstructured.Unstructured{}
+		nodeclass.SetGroupVersionKind(ncGVK)
+		builder.Watches(
+			nodeclass,
+			nodeclaimutil.NodeClassEventHandler(c.kubeClient),
+		)
```

Comment:
Don't you need to re-set the builder here? like 
builder := builder.Watches()

Reply:
`builder` is a pointer that is updated by `watches`, since `watches` is updates the pointer the returned values will be the same as the value we have at the top level 
- https://github.com/kubernetes-sigs/controller-runtime/blob/a17fd5870b906b50662fed6658265f5f8cf0863e/pkg/builder/controller.go#L137
- https://github.com/kubernetes-sigs/controller-runtime/blob/a17fd5870b906b50662fed6658265f5f8cf0863e/pkg/builder/controller.go#L179

Reply:
Good practice to probably re-set the builder anyway.